### ISSUE TITLE
Mod Evasive Not Working for 100 blacklisted URI requests

### ISSUE THREAD

**Issue by abhishekjadhav4** [jvdmr/mod_evasive#24]
Hello @jvdmr 
I am using the following settings, when I test it with 10 requests it works fine. On changing it with 100 requests it is not working.
FYI, I am testing it with a shell script having curl POST request in a loop.
<IfModule mod_evasive.c>
    DOSEnabled          true
    DOSHashTableSize  3097
    DOSPageCount        100
    DOSSiteCount        10000000
    DOSPageInterval     3600
    DOSSiteInterval     3600
    DOSBlockingPeriod   86400
    DOSWhitelistUri ^((?!j_spring_security_check).)*$
</IfModule>

**Comment by jvdmr** [comment]
I'm afraid your usecase is not covered by mod_evasive, and you're running into limitations caused by Apache's architecture. Specifically, Apache has several worker/listener/child processes, and they each have their own runtime memory and thus their own tally of 'hits'. If you keep hitting the same worker, you will run into the limit exactly when you'd expect, but that's very unlikely. Workers are killed and replaced after a while, and the tally does not transfer into a new worker.

If you keep the configured numbers low, that doesn't matter very much because soon enough all available workers will have blocked the attacker, but with high numbers that can take quite a bit longer.

For example, with your settings on an otherwise clean Apache installation, I see blocked requests starting after about 120 requests, but some requests keep getting through up until about 430 requests.

In conclusion, mod_evasive is meant to block heavy loads of traffic from a single source that try to bring a server down by overloading it. It is **not** meant to stop someone trying to break into your system by guessing passwords or anything like that.
